### PR TITLE
Using http-middleware interfaces instead of psr-15 draft

### PR DESCRIPTION
**Pull request by dirx**
* using https://github.com/http-interop/http-middleware instead of our own PSR-15 Draft extracted Interfaces
* Stack now delegates
* Stack now has it´s own StackInterface

This simplifies Stack a lot but needs further discussion whether Stack should support DelegateInterface or MiddlewareInterface or both.

### CODE REVIEW

**Comment by dirx** [comment]
@zavalit shed some light

File: src/StackInterface.php [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-use Psr\Http\Message\RequestInterface;
-use Psr\Http\Message\ResponseInterface;
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface;
 
-interface StackInterface
+interface StackInterface extends DelegateInterface
 {
     /**
      * Return an instance with the specified middleware added to the stack.
```
Comment (zavalit) [inline src/StackInterface.php]: if that Interop guys don't use anything but ```process(ServerRequestInterface $request)```. I honestly don't see any usecase for ```withMiddleware(MiddlewareInterface $middleware)``` to proccess a Stack. May be by adding middlewares on runtime, but what could be good for?

Comment (zavalit) [inline src/StackInterface.php]: i would just stay by current implementation a just change methods ```withMiddleware/withoutMiddleware``` visibility to ```private``` and get rid  of ```StackInterface```. what do you think?

Comment (dirx) [inline src/StackInterface.php]: I totally agree. For adding middlewares on runtime `withMiddleware(MiddlewareInterface $middleware)` does not really do a good job - it just appends middleware. There are better ways like conditional middlewares or simple routing (eg. via a fastroute router) to other stacks.

Let´s drop `StackInterface` and implement `DelegateInterface` and cleanup the withs ...

I will update the PR.

**Pull request merged**